### DISCUSSION TITLE
Chunk Statistics Tab?

### DISCUSSION THREAD

**Discussion by Blackpaw** [moosefs/moosefs#422] [Ideas]
Would it be possible to get a chunk statistics tab, similar to LizardFS's? I find it very useful for a quick overview of the system and for highlighting issues and repair progress.

![image](https://user-images.githubusercontent.com/65111/112231117-ce047300-8c81-11eb-840d-5901ef368544.png)

**Comment by oszafraniec** [comment]
What's wrong with current mfs.cgi? 🙄

https://moosefs.com/blog/how-install-moosefs/
![CGI_main_view](https://user-images.githubusercontent.com/3938516/112279740-892e0b80-8c84-11eb-8706-bd1d788a15fa.jpg)

**Reply by Blackpaw** [reply]
I find it busy and hard to parse with a glance, and unless I'm mistaken (which I may be), it lacks the detail of chunk replication per goal.

nb. I already have a 3 node moose cluster setup and am migrating my lizardfs data to it, so no need for the install link :)

**Reply by oszafraniec** [reply]
Let me do it for you ;) It's a matrix. On the left chunks with goal. On the top chunks with copies. If anything is not green, you see why and how much. It's simple and i find Lizard CGI to complicated, too much to read :) 

250k chunks has only 3 copies and should have 4. 78 chunks are OK. Mayby someone changed the goal on files, can be.
<img width="731" alt="Zrzut ekranu 2021-03-24 o 12 00 28" src="https://user-images.githubusercontent.com/3938516/112299990-a8cf2f00-8c98-11eb-83c0-522354e57ef2.png">

418 chunks are overgoal, not big deal, will go away in a mean time
<img width="1215" alt="Zrzut ekranu 2021-03-24 o 12 01 05" src="https://user-images.githubusercontent.com/3938516/112299996-aa005c00-8c98-11eb-9a99-d65c2a3e548c.png">

Something is strange, files from different goals are undergoal, mayby drive failed?
<img width="673" alt="Zrzut ekranu 2021-03-24 o 12 02 51" src="https://user-images.githubusercontent.com/3938516/112300219-e469f900-8c98-11eb-8099-e1b1f86557ee.png">

**Reply by Blackpaw** [reply]
Thanks, that makes more sense.

My moose cluster is relatively new, I haven't encountered any failures yet - Guess I could kill a chunkserver to see what it looks like :)

**Reply by oszafraniec** [reply]
You can easly simulate that by changing goals/classes up and down and see how cluster reacts. Restarting chunkserver or killing it also can do the trick ;) Play with it, get used to and forget lizard :D

**Comment by xandrus** [comment]
Personally, I believe that MooseFS CGI is much better at tracing the problem in a cluster.
All the most important information about chunks are always available at the bottom of "All chunks state matrix".
![image](https://user-images.githubusercontent.com/13887484/112285962-255b1100-8c8b-11eb-82ba-29c5260c9ffd.png)

**Reply by Blackpaw** [reply]
Maybe I just need to get used to the new layout - familiarity and all. I'll try a few failure domains to see how it goes.
